lib/chutney/TorNet.py

```diff
@@ -1208,17 +1212,33 @@ def getLastBootstrapStatus(self):
                 if m:
                     percent, keyword, message = m.groups()
                     percent = int(percent)
-        return (percent, keyword, message)
+        self.most_recent_bootstrap_status = (percent, keyword, message)
+
+    def getLastBootstrapStatus(self):
+        """Return the last bootstrap message fetched by
+           updateLastBootstrapStatus as a 3-tuple of percentage
+           complete, keyword (optional), and message.
+        """
+        return self.most_recent_bootstrap_status
+
+    def updateLastStatus(self):
+        """Update last messages this node has received, for use with
+           isBootstrapped and the getLast* functions.
+        """
```

Comment:
What ensures that this function is called before getLastBootstrapStatus and getLastOnionServiceDescStatus?  IMO we should document that this function needs to be called before those functions will return anything, and document why we're confident that it was called first.

Reply:
Added commentary on how those functions interact, thanks for catching that.

If there's a better way to clean up the code in this area than how I'm doing it, I'd like to know, because having separate functions to effect and recall state seems a tad ugly, but I believe it is still better than the original way of adding statuses to lists, then passing as arg to a function already with enough args, and then `zip(controllers, statuses)`. And it seems to make the code a bit more DRY overall.